android/app/src/main/java/org/fossasia/openevent/viewmodels/LocationActivityViewModel.java

```diff
@@ -0,0 +1,76 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.LiveRealmData;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+
+import java.util.List;
+import java.util.Locale;
+
+
+import io.reactivex.Observable;
+
+import timber.log.Timber;
+
+public class LocationActivityViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private LiveData<List<Session>> session;
+    private MutableLiveData<List<Session>> filteredSession;
+    private String searchText = "";
+
+    public LocationActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Session>> getSessionByLocation(String location, String searchText) {
+        setSearchText(searchText);
+        if (filteredSession == null)
+            filteredSession = new MutableLiveData<>();
+        if (session == null) {
+            session = new MutableLiveData<>();
+            LiveRealmData<Session> sessionLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getSessionsByLocation(location));
+            session= Transformations.map(sessionLiveRealmData, input -> input);
+            getFilteredSessions();
+        } else {
+            getFilteredSessions();
+        }
+        return filteredSession;
+    }
```

Comment:
Logic here is too complex, maybe wait for another PR which encapsulates such functionality through a wrapper

Reply:
How about leaving the filtering functionality upto the adapter for now and creating a new issue for it? Then this PR can be merged for the time being.

Reply:
Actually this functionality is the same as in the filter method in the adapter.

Reply:
See other PRs which have handled filtering. I know this is a drag, but if we leave things incomplete, there can be months until someone takes that up

Reply:
I have had a look at them. Almost all that have been merged have left the filtering work upto the adapter only.

Reply:
That's not true. You may be looking at the older PRs. Let me find one which specifically addresses filtering in ViewModel. @srv-twry Can you help?

Reply:
@Kartik2016 There were only two PRs that were merged without addressing filtering in ViewModel. Both of them were mine and they where merged before we realised that the filtering was being done inside the Adapter.
After the issue was realised i created issues for both the merged PRs and one of those issues was resolved today itself.
Now you can go ahead and use the ```FilterableRealmLiveData``` that has been created to ease the process of filtering.

Reply:
Ok i'll work on this.